flink-yarn/src/main/java/org/apache/flink/yarn/AbstractYarnClusterDescriptor.java

```diff
@@ -164,6 +164,12 @@ public AbstractYarnClusterDescriptor() {
 				throw new RuntimeException("Unable to locate configuration file in " + confFile);
 			}
 			flinkConfigurationPath = new Path(confFile.getAbsolutePath());
+
+			if (flinkConfiguration.containsKey(ConfigConstants.YARN_VCORES)) {
+				slots = flinkConfiguration.getInteger(ConfigConstants.YARN_VCORES, -1);
+			} else if (flinkConfiguration.containsKey(ConfigConstants.TASK_MANAGER_NUM_TASK_SLOTS)) {
+				slots = flinkConfiguration.getInteger(ConfigConstants.TASK_MANAGER_NUM_TASK_SLOTS, -1);
+			}
```

Comment:
I think we should make the behaviour consistent with Mesos and standalone where we set `slots = flinkConfiguration.getInteger(ConfigConstants.TASK_MANAGER_NUM_TASK_SLOTS, 1)`.